### PR TITLE
feat: Lick of paint for data warehouse

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Waiting on the CI - noticed a comment about collapsable data tables and realised every time I look at the data warehouse page my design consistency mind screams out. So I scratched the itch.

## Changes

|Before|After|
|----|----|
| Screenshot 2024-05-23 at 21 53 40 | Screenshot 2024-05-23 at 21 53 09 |
| Screenshot 2024-05-23 at 21 56 16 | Screenshot 2024-05-23 at 21 56 49 |

* Borders and padding and radius fixes
* Changed the tree items to be standard LemonButtons
* Made it collapsable
* Fixed up some of the wrapping things
* Changed some buttons to secondary (there should only ever really be one primary...

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx [hunk 1]
```diff
@@ -20,13 +16,12 @@ export interface TreeRowProps {
 export function TreeRow({ item, selected }: TreeRowProps): JSX.Element {
     return (
         <li>
-            <div className={clsx('TreeRow text-ellipsis cursor-default', selected ? 'TreeRow__selected' : '')}>
```
Comment (EDsCODE) [inline frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx:23]: Smart. Main reason I avoided button at first though was information density. The buttons even xsmall have some space that ruins the dense editor feel

**Review by EDsCODE — APPROVED** [review]
Just a quick note on density. I briefly looked to see if we had it customizeable but looks like LemomButton__chrome is set in there. No worries. I'll come back to it if the density ever becomes an issue

Thanks for the refresh. this is awesome!

**Comment by zlwaterfield** [comment]
This is beautiful. Benterprise to the rescue.

File: frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx:23]: There is actually a `noPadding` option that we could add if we want to have a bit more control over it. That way we still get the other niceties of the button, but with tighter layout

**Pull request merged**